nixos/modules/services/networking/go-neb.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.go-neb;
+
+  configFile = pkgs.writeText "config.yml" (builtins.toJSON cfg.config);
+in {
+  options.services.go-neb = {
+    enable = mkEnableOption "Extensible matrix bot written in Go";
+
+    bindAddress = mkOption {
+      type = types.str;
+      description = "Port (and optionally address) to listen on.";
+      default = ":4050";
+    };
+
+    baseUrl = mkOption {
+      type = types.str;
+      description = "Public-facing endpoint that can receive webhooks.";
+    };
+
+    config = mkOption {
```

Comment:
Any interest in providing a mechanism for handling secrets in here? From the looks of the format likely not trivial to add great support...

Reply:
The only secret worth protecting would be `clients.*.AccessToken`. I'm not sure how I would keep that out of the nix store when the program does not support reading secrets from dedicated files.

Reply:
Yeah the solution wouldn't be pretty. Some modules will read the password on startup and then copy+paste+modify the generated configuration file replacing any secrets in a protected directory. Using `DynamicUser` adds additional complications. Secrets management in NixOS is not an easy thing, unfortunately.